packages/intl-locale/index.ts

```diff
@@ -695,6 +757,43 @@ export class Locale {
     }
   }
 
+  /**
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo
+   * https://tc39.es/proposal-intl-locale-info/#sec-Intl.Locale.prototype.getWeekInfo
+   */
+  public getWeekInfo() {
+    try {
+      const info = Object.create(Object.prototype)
```

Comment:
missing `initializedLocale` check per spec

Reply:
I'm not sure I understand what this means:

In the spec here, I don't see any `initializedLocale` check:
<img width="699" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/de9551b2-be84-47cd-bcf4-5b4fa7028786">

Is it the `RequireInternalSlot`?

Or is it the step `Assert: locale matches the unicode_locale_id production.` within `WeekInfoOfLocale(loc)`?

<img width="851" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/1001e9c8-35ca-4bf1-8ee3-e9f1b0d25225">

Reply:
Isn't the `initializedLocale` ensured by the constructor of the Locale?

<img width="311" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/23e9218f-50d7-4728-b28d-9d81ebbf2c1a">

Reply:
Also if you check `baseName`, there is the same `RequireInternalSlot` in the spec:
<img width="785" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/918b82c7-601b-4b96-bbff-624f446cfc8f">

But the implem only calls `getInternalSlots(this)`:

<img width="480" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/03b770de-5e97-4ff0-974e-9e60d605a755">

Which is also what `weekInfoOfLocale` does internally:
<img width="364" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/2f57bff0-2aa6-4b5b-9cae-1e08cc1095d7">
<img width="451" alt="image" src="https://github.com/formatjs/formatjs/assets/22725671/f0098588-4180-46d0-8933-67a292517c21">

So I'm not sure what I need to add to this method

Reply:
I think you basically just need to make sure `initializedLocale` is not `undefined` in `internalSlots`, otherwise throw per `RequireInternalSlot` spec

Reply:
Added 👍